File: Belajar/ProgramATM_SwitchCase/SwitchATM.php
<?php

while (true) {
    $nama = readline("Masukkan nama anda: ");

    $konfirmasi = "";
    while ($konfirmasi != "true" && $konfirmasi != "false") {
        $konfirmasi = strtolower(readline("Apakah nama sudah benar? (true/false): "));
    }

    if ($konfirmasi == "true") break;
    echo "Silakan masukkan nama kembali.\n";
}

$nimBenar = false;
while (!$nimBenar) {
    $nim = readline("Masukkan NIM: ");

    if ($nim !== "" && ctype_digit($nim)) {
        $nimBenar = true;
    } else {
        echo "NIM harus angka!\n";
    }
}

$saldo = intval($nim);

echo "\nNama: $nama\n";
echo "Saldo Awal: Rp " . number_format($saldo, 0, ',', '.') . "\n\n";

while (true) {
    echo "===== MENU ATM =====\n";
    echo "1. Cek Saldo\n";
    echo "2. Tarik Tunai\n";
    echo "3. Setor Tunai\n";
    echo "4. Transfer\n";
    echo "5. Keluar\n";

    $pilihan = readline("Pilih menu (1-5): ");
    echo "\n";

    switch ($pilihan) {
        case "1":
            echo "Saldo: Rp " . number_format($saldo, 0, ',', '.') . "\n\n";
            break;

        case "2":
            $tarik = readline("Masukkan jumlah tarik tunai: ");
            if (!ctype_digit($tarik) || $tarik <= 0) {
                echo "Input tidak valid!\n\n";
            } elseif ($tarik > $saldo) {
                echo "Saldo tidak cukup!\n\n";
            } else {
                $saldo -= $tarik;
                echo "Saldo baru: Rp " . number_format($saldo, 0, ',', '.') . "\n\n";
            }
            break;

        case "3":
            $setor = readline("Masukkan jumlah setor tunai: ");
            if (ctype_digit($setor) && $setor > 0) {
                $saldo += $setor;
                echo "Saldo baru: Rp " . number_format($saldo, 0, ',', '.') . "\n\n";
            } else {
                echo "Input tidak valid!\n\n";
            }
            break;

        case "4":
            $transfer = readline("Masukkan nominal transfer: ");
            if (!ctype_digit($transfer) || $transfer <= 0) {
                echo "Input tidak valid!\n\n";
            } elseif ($transfer > $saldo) {
                echo "Saldo tidak cukup!\n\n";
            } else {
                $rek = readline("Masukkan nomor rekening tujuan: ");
                $saldo -= $transfer;
                echo "Transfer berhasil ke $rek\n";
                echo "Saldo baru: Rp " . number_format($saldo, 0, ',', '.') . "\n\n";
            }
            break;

        case "5":
            echo "Terima kasih!\n";
            exit;

        default:
            echo "Pilihan tidak valid!\n\n";
    }
}
?>

File: Belajar/ProgramATM_SwitchCase/SwitchATM.cpp
#include <iostream>
#include <string>
using namespace std;

bool angkaValid(string s) {
    if (s == "") return false;
    for (char c : s) {
        if (c < '0' || c > '9') return false;
    }
    return true;
}

int main() {
    string nama;
    string konfirmasi;

    while (true) {
        cout << "Masukkan nama anda: ";
        getline(cin, nama);

        konfirmasi = "";
        while (konfirmasi != "true" && konfirmasi != "false") {
            cout << "Apakah nama sudah benar? (true/false): ";
            getline(cin, konfirmasi);
        }

        if (konfirmasi == "true") break;
        cout << "Silakan masukkan nama kembali.\n";
    }

    string nim;
    while (true) {
        cout << "Masukkan NIM: ";
        getline(cin, nim);

        if (angkaValid(nim)) break;
        cout << "NIM harus angka!\n";
    }

    long long saldo = stoll(nim);

    cout << "\nNama: " << nama << endl;
    cout << "Saldo Awal: Rp " << saldo << "\n\n";

    while (true) {
        cout << "===== MENU ATM =====\n";
        cout << "1. Cek Saldo\n";
        cout << "2. Tarik Tunai\n";
        cout << "3. Setor Tunai\n";
        cout << "4. Transfer\n";
        cout << "5. Keluar\n";
        cout << "Pilih menu (1-5): ";

        string pilih;
        getline(cin, pilih);
        cout << "\n";

        if (pilih == "1") {
            cout << "Saldo: Rp " << saldo << "\n\n";
        }

        else if (pilih == "2") {
            string tarik;
            cout << "Masukkan jumlah tarik tunai: ";
            getline(cin, tarik);

            if (angkaValid(tarik)) {
                long long t = stoll(tarik);
                if (t <= saldo) {
                    saldo -= t;
                    cout << "Saldo baru: Rp " << saldo << "\n\n";
                } else {
                    cout << "Saldo tidak cukup!\n\n";
                }
            } else {
                cout << "Input tidak valid!\n\n";
            }
        }

        else if (pilih == "3") {
            string setor;
            cout << "Masukkan jumlah setor: ";
            getline(cin, setor);

            if (angkaValid(setor)) {
                saldo += stoll(setor);
                cout << "Saldo baru: Rp " << saldo << "\n\n";
            } else {
                cout << "Input tidak valid!\n\n";
            }
        }

        else if (pilih == "4") {
            string transfer;
            cout << "Masukkan nominal transfer: ";
            getline(cin, transfer);

            if (angkaValid(transfer)) {
                long long tr = stoll(transfer);
                if (tr <= saldo) {
                    string rek;
                    cout << "Masukkan nomor rekening tujuan: ";
                    getline(cin, rek);

                    saldo -= tr;
                    cout << "Transfer berhasil ke " << rek << endl;
                    cout << "Saldo baru: Rp " << saldo << "\n\n";
                } else {
                    cout << "Saldo tidak cukup!\n\n";
                }
            } else {
                cout << "Input tidak valid!\n\n";
            }
        }

        else if (pilih == "5") {
            cout << "Terima kasih!\n";
            return 0;
        }

        else {
            cout << "Pilihan tidak valid!\n\n";
        }
    }
}
